apps/aechannel/src/aesc_fsm.erl

```diff
@@ -841,9 +882,29 @@ awaiting_signature(cast, {?SIGNED, ?SHUTDOWN_ACK, SignedTx} = Msg,
             D1 = send_shutdown_ack_msg(NewSignedTx, D),
             D2 = D1#data{latest = undefined,
                         log    = log_msg(rcv, ?SIGNED, Msg, D1#data.log)},
-            report(on_chain_tx, NewSignedTx, D1),
+            report_on_chain_tx(close_mutual, NewSignedTx, D1),
             close(close_mutual, D2)
         end, D);
+awaiting_signature(cast, {?SIGNED, close_solo_tx, SignedTx} = Msg,
```

Comment:
I am surprised that this clause does not store `SignedTx` for `aesc_offchain_state:get_latest_signed_tx` in `open(enter, _OldSt, D) ->` to find it.